libbpf-rs/src/ringbuf.rs

```diff
@@ -136,7 +136,12 @@ impl<'a> RingBuffer<'a> {
     pub fn poll(&self, timeout: Duration) -> Result<()> {
         assert!(!self.ptr.is_null());
 
-        let ret = unsafe { libbpf_sys::ring_buffer__poll(self.ptr, timeout.as_millis() as i32) };
+        let mut timeout_ms = -1;
+        if timeout != Duration::MAX {
+            timeout_ms = timeout.as_millis() as i32;
```

Comment:
looks good, but let's update the doc comment for poll() to specify that Duration::MAX is used to mean "infinite timeout"?